tests/serverpod_test_server/test_integration/test_tools/start_timeout_test/start_timeout_test.dart

```diff
@@ -0,0 +1,78 @@
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+import 'package:test/test.dart';
+
+enum TestTag { include, exclude }
+
+void main() async {
```

Comment:
Question: Arn't these tests going to be pretty slow? Any possibility to get around the 4 second wait and still timeout on the right thing?

Is the (last) passing test required for example?

Reply:
Yes, this is not good, especially since we're running tests with `concurrency=1`. 

I looked at the [`fake_async` package](https://pub.dev/packages/fake_async), but for some reason this doesn't actually work: 
```dart
  fakeAsync((async) async {
    withServerpod(
      'Given that runMode is set to staging when calling withServerpod',
      // The staging config has intentionally been set to not find the database and provoke timeout.
      runMode: ServerpodRunMode.staging,
      serverpodStartTimeout: Duration(seconds: 20),
      (sessionBuilder, endpoints) {
        test(
            'then dummy test should not be run and exit code should be non-zero',
            () async {
          expect(true, true);
        });
      },
    );

    async.elapse(Duration(seconds: 20));
  });
```

But I think the issue here might the timing of when we're calling `await Future.delayed(const Duration(seconds: 10));` internally in the database connection logic. Especially since we're doing that periodically, so a single `async.elapse` wouldn't work anyway probably. I tried adding in some `await Future.delayed` before the elapse to see if that did some difference but no success.

What we could do is maybe tag these slow tests and run them concurrently to the other integration tests?

Reply:
> Is the (last) passing test required for example?

It's more for completeness, since we don't have a passing test with that config anywhere. The passing test it really quick compared to the others though.

Reply:
Discussed IRL: there isn't much to do regarding speeding up the time. In a separate PR, we will separate these test that are long running so that they can be run in parallel.